prov/usnic/src/usdf_cq.c

```diff
@@ -610,9 +611,10 @@ static const char *
 usdf_cq_strerror(struct fid_cq *eq, int prov_errno, const void *err_data,
 		 char *buf, size_t len)
 {
-	strncpy(buf, "CQ Error", len-1);
-	buf[len-1] = '\0';
-	return buf;
+	if (buf && len) {
+		return strncpy(buf, strerror(-prov_errno), len);
```

Comment:
From `man 3 strncpy`:
>  Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null terminated.

`strncpy` won't null-terminate `buf` if the result of `strerror(-prov_errno)` is longer than `len`.  It would be good to retain the `buf[len-1] = '\0'` statement (obviously inside of the `if` that checks for `len != 0`).

We should probably use `fi_strerror` instead of `strerror` here.

I think that the `prov_errno` that will be passed in will be positive, not negative, so you shouldn't be negating it when passing to `strerror`/`fi_strerror`.